lib/puppet/provider/package/pip.rb

```diff
@@ -166,49 +188,79 @@ def latest_with_old_pip
       execpipe command_and_options do |process|
         process.collect do |line|
           # PIP OUTPUT: Using version 0.10.1 (newest of versions: 1.2.3, 4.5.6)
-          if line =~ /Using version (.+?) \(newest of versions/
-            return $1
+          if line =~ /Using version .+? \(newest of versions: (.+?)\)/
+            textAfterLastMatch = $1.chomp(")\n")
+            versionList = textAfterLastMatch.split(', ').sort do |x,y|
+              Puppet::Util::Package.versioncmp(x, y)
+            end
+            return versionList
```

Comment:
Before, `latest_with_old_pip` was returning the version without sorting avaialable versions. Will this change the behaviour, now that we are sorting the versions